tools/ci/pr_preview.py

```diff
@@ -0,0 +1,403 @@
+#!/usr/bin/env python
+
+# The service provided by this script is not critical, but it shares a GitHub
+# API request quota with critical services. For this reason, all requests to
+# the GitHub API are preceded by a "guard" which verifies that the subsequent
+# request will not deplete the shared quota.
+#
+# In effect, this script will fail rather than interfere with the operation of
+# critical services.
+
+import argparse
+import json
+import logging
+import os
+import subprocess
+import time
+
+import requests
+
+# The ratio of "requests remaining" to "total request quota" below which this
+# script should refuse to interact with the GitHub.com API
+API_RATE_LIMIT_THRESHOLD = 0.2
+# The GitHub Pull Request label which indicates that a Pull Request is expected
+# to be actively mirrored by the preview server
+LABEL = 'safe for preview'
```

Comment:
Nit; the generic name `LABEL` and `HasLabel` functions mislead me a bit when reading - it required me to come up to this definition and see that the label is the one that marks it as safe to mirror. Maybe just renaming `HasLabel` to `HasMirroringLabel` would suffice to remove my confusion.

Reply:
I considered parameterizing the label name, but that seemed a little silly given that we only expect one value. Renamed the function as you requested.